modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/LZ4CompressionTestCase.java

```diff
@@ -0,0 +1,174 @@
+     *
+     * @throws AndesClientConfigurationException
+     * @throws IOException
+     * @throws JMSException
+     * @throws NamingException
+     * @throws AndesClientException
+     */
+    @Test(groups = "wso2.mb", description = "Compressed message content validation test case")
+    public void performQueueContentSendCompressReceiveTestCase() throws AndesClientConfigurationException, IOException,
+            JMSException, NamingException, AndesClientException, XPathExpressionException {
+
+        // Reading message content
+        char[] inputContent = new char[SIZE_TO_READ];
+
+        BufferedReader inputFileReader = new BufferedReader(
+                new FileReader(AndesClientConstants.MESSAGE_CONTENT_INPUT_FILE_PATH_WITHOUT_REPETITIONS_256KB));
+        inputFileReader.read(inputContent);
+
+        // Creating a consumer client configuration
+        AndesJMSConsumerClientConfiguration consumerConfig = new AndesJMSConsumerClientConfiguration(getAMQPPort(),
+                ExchangeType.QUEUE, "QueueContentSendCompressReceive");
+        consumerConfig.setMaximumMessagesToReceived(EXPECTED_COUNT);
+        // writing received messages.
+        consumerConfig.setFilePathToWriteReceivedMessages(AndesClientConstants.FILE_PATH_TO_WRITE_RECEIVED_MESSAGES);
+        consumerConfig.setAsync(false);
+
+        // Creating a publisher client configuration
+        AndesJMSPublisherClientConfiguration publisherConfig = new AndesJMSPublisherClientConfiguration(getAMQPPort()
+                , ExchangeType.QUEUE, "QueueContentSendCompressReceive");
+
+        publisherConfig.setNumberOfMessagesToSend(SEND_COUNT);
+        // message content will be read from this path and published
+        publisherConfig.setReadMessagesFromFilePath(AndesClientConstants
+                .MESSAGE_CONTENT_INPUT_FILE_PATH_WITHOUT_REPETITIONS_256KB);
+
+        // Creating clients
+        AndesClient consumerClient = new AndesClient(consumerConfig, true);
+        consumerClient.startClient();
+
+        AndesClient publisherClient = new AndesClient(publisherConfig, true);
+        publisherClient.startClient();
+
+        AndesClientUtils.waitForMessagesAndShutdown(consumerClient, AndesClientConstants.DEFAULT_RUN_TIME);
+
+        // Reading received message content
+        char[] outputContent = new char[SIZE_TO_READ];
+
+        BufferedReader outFileReader =
+                new BufferedReader(new FileReader(AndesClientConstants.FILE_PATH_TO_WRITE_RECEIVED_MESSAGES));
+        outFileReader.read(outputContent);
+
+        // Evaluating
+        Assert.assertEquals(publisherClient.getSentMessageCount(), SEND_COUNT, "Message sending failed.");
+        Assert.assertEquals(consumerClient.getReceivedMessageCount(), EXPECTED_COUNT, "Message receiving failed.");
+        Assert.assertEquals(new String(outputContent), new String(inputContent), "Message content has been modified.");
```

Comment:
Is using string compare method of String class and assert the boolean more appropriate?